db/active_replicator_checkpointer.go

```
@@ -692,80 +692,83 @@ func (c *Checkpointer) setRetry(checkpoint *replicationCheckpoint, setFn setChec
-		return
-	}
-	if checkpoint == nil {
-		checkpoint = &replicationCheckpoint{}
+		base.WarnfCtx(ctx, "Unable to retrieve local status doc for %s, status not updated", clientID)
+		return "", nil
+	}
+	if currentStatus == nil {
+		currentStatus = &ReplicationStatusDoc{
+			Status: &ReplicationStatus{
+				ID: clientID,
+			},
+		}
 	}
 
-	if checkpoint.Status == nil {
-		checkpoint.Status = &ReplicationStatus{}
+	currentStatus.Status.Status = status
+	currentStatus.Status.ErrorMessage = errorMessage
+	if pushStats != nil {
+		currentStatus.Status.PushReplicationStatus = *pushStats
+	}
+	if PullStats != nil {
+		currentStatus.Status.PullReplicationStatus = *PullStats
 	}
 
-	checkpoint.Status.Status = status
-	checkpoint.Status.ErrorMessage = errorMessage
-	base.TracefCtx(ctx, base.KeyReplicate, "setLocalCheckpoint(%v)", checkpoint)
-	newRev, putErr := putSpecial(dataStore, DocTypeLocal, CheckpointDocIDPrefix+clientID, checkpoint.Rev, checkpoint.AsBody(), localDocExpirySecs)
-	if putErr != nil {
-		base.WarnfCtx(ctx, "Unable to persist status in local checkpoint for %s, status not updated: %v", clientID, putErr)
-	} else {
-		base.TracefCtx(ctx, base.KeyReplicate, "setLocalCheckpointStatus successful for %s, newRev: %s: %+v %+v", clientID, newRev, checkpoint, checkpoint.Status)
+	return putSpecial(dataStore, DocTypeLocal, ReplicationStatusDocIDPrefix+clientID, currentStatus.Rev, currentStatus.AsBody(), localDocExpirySecs)
+}
+
+// removeLocalStatus removes a replication status.
+func removeLocalStatus(ctx context.Context, dataStore base.DataStore, clientID string) (err error) {
+	base.TracefCtx(ctx, base.KeyReplicate, "removeLocalStatus()")
+
+	// getLocalStatus to obtain the current rev
+	currentStatus, err := getLocalStatus(ctx, dataStore, clientID, 0)
+	if err != nil {
+		base.WarnfCtx(ctx, "Unable to retrieve local status doc for %s, status not updated", clientID)
+		return nil
+	}
```

this is not going to return nil if `KeyNotFound` so should we return an error here? 

What situations do we expect this to happen?